### PR TITLE
API for search within time range.

### PR DESCRIPTION
**Pull request by vivek-sethia**

### CODE REVIEW

File: app/stores/sighting.js [hunk 1]
```diff
@@ -94,5 +94,43 @@ module.exports = {
...
+            fromTs = new Date(req.ts),
+            toTs;
+
+        let rangeValue = parseInt(range, 10),
+            rangeSpan = range.replace(/\d+/g, '');
+
+        switch(rangeSpan) {
+            // week
+            case 'w':
+                toTs = new Date(fromTs.getTime() + rangeValue * 7 * 24 * 60 * 60 * 1000);
```
Comment (samitsv) [inline app/stores/sighting.js:113]: maybe change the numbers to some constant values in config file? :)

Comment (vivek-sethia) [reply app/stores/sighting.js:113]: These values are not configurable and these are used for time calculation, so I think it's not required to put into config file.

File: app/api/sighting.js [hunk 2]
```diff
@@ -205,5 +205,46 @@ module.exports = {
...
+        logger.info('Get Pokemon Sightings within a specific time range');
+
+        sighting.getByTimeRange(req.params, function(success, message) {
+            if(success === 1)
+                res.status(200).json({message: 'Success', data: message});
+            else
+                res.status(404).json({message: 'Failure. No sighting details within this time range exists!', data: message});
+        });
     }
 };
```
Comment (swathi-ssunder) [inline app/api/sighting.js:250]: Add GetSightingByTimeRange to apidoc.json. Else, it will not be displayed.

File: app/stores/sighting.js (hunk 1 above)
Comment (swathi-ssunder) [reply app/stores/sighting.js:113]: This is fine for me too. Since they are time conversion values, need not be in the config.

**Pull request merged**